hooks.js

```diff
@@ -1,15 +1,51 @@
 export class Hooks {
     constructor() {
-        this.recognizer = new SequenceRecognizer();
+        this.seq_key_recognizer = new SequenceRecognizer();
+        this.seq_code_recognizer = new SequenceRecognizer();
+        this.chord_key_recognizer = new ChordRecognizer();
+        this.chord_code_recognizer = new ChordRecognizer();
         this.pressed_keys = new Set();
+        this.pressed_codes = new Set();
     }
 
-    add_reset_sequence(path) {
-        this.recognizer.add_reset_sequence(path);
+    add_reset_key_sequence(path) {
+        this.seq_key_recognizer.add_reset_sequence(path);
     }
 
-    add_sequence(path, callback) {
-        this.recognizer.add_sequence(path, callback);
+    add_reset_code_sequence(path) {
```

Comment:
Change these functions from `add_reset_xxx_sequence` -> add_reset_sequence('xxx', path, callback), this will be easier to remember.